### PR TITLE
Reorganize topics in the Fleet and Elastic Agent docs

### PR DESCRIPTION
**Pull request by dedemorton**
Improves organization of the Fleet User Guide. This is a fairly major reorganization with many content changed, moved content, and renamed files.

*Please note that the steps for adding a Fleet Server are updated in https://github.com/elastic/observability-docs/pull/1091, but not yet merged.*

Preview: https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/index.html 

During your review:

1. Please comment on the overall structure and TOC.
2. The diff is a little hard to follow, so I'll list topics with meaningful content changes here so you can review them carefully:

- [fleet-quick-start.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/fleet-quick-start.html)  (Step 2: Add an Elastic Agent to Fleet)
- [fleet-quick-start-traces.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/fleet-quick-start-traces.html) (Step 2: Add an Elastic Agent to Fleet)
- [integrations.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/integrations.html) (intro - stop at table)
- [elastic-agent-installation.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/elastic-agent-installation.html) (all)
- [install-fleet-managed-elastic-agent.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/install-fleet-managed-elastic-agent.html) (all)
- [nstall-elastic-agents-in-containers.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/install-elastic-agents-in-containers.html) (all)
- [manage-agents-in-fleet.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/manage-agents-in-fleet.html) (all)
- [manage-agents.html](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/manage-agents.html) (all)
- [troubleshooting-intro.html](http://localhost:8000/guide/troubleshooting-intro.html) (all)
- [fleet-troubleshooting.html](http://localhost:8000/guide/fleet-troubleshooting.html) (just the intro table)

Remaining TODOs:

-  [x] Remove xpack role
-  [x] Confirm that installation docs are still correct after moving sections around. (DeDe)
    -  [x] Logs/metrics/uptime quick start
    - [x] Fleet steps
    - [x] Standalone steps
-  [x] Add installation layout topic for standalone agents (or make existing one more generic)
-  [x] Check overall flow.

TODO after merging:
-  Move the quick starts to the observability guide (after this PR is merged)
-  Open PR in Kibana to update in-product doc link: "See the Elastic Agent docs for RPM / DEB deploy instructions."
-  Make the steps and component descriptions more consistent across pages.

Note that the TODOs in the source files are for future work, not necessarily for this review.

### CODE REVIEW

**Review by bmorelli25 — COMMENTED** [review]
Huge improvement IMO. Initial thoughts:

1. Are we still planning on moving the quick start guides? If so, I'm guessing that will be handled in a separate PR?
2. In the installation bucket, you put Fleet before standalone. I think it also makes sense to have `Centrally manage Elastic Agents in Fleet` come before `Elastic Agent standalone configuration`.
3. At the end of the [`Install standalone Elastic Agent`](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/install-standalone-elastic-agent.html) topic, there's a link: `Refer to Installation layout for the location of installed Elastic Agent files`. This link sends users to the [Fleet install topic](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/install-fleet-managed-elastic-agent.html#installation-layout). I think this could be confusing. What if we used includes to have the installation layout topic in both files?
4. Similarly, at the end of the [`Install standalone Elastic Agent`](https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/install-standalone-elastic-agent.html) topic, there's a link to the Elastic Agent standalone configuration docs as a next step. I think a similar link to the Fleet content would be useful at the end of the Fleet install topic.

**Comment by dedemorton** [comment]
Thanks @bmorelli25! In response to your questions:

>Are we still planning on moving the quick start guides? If so, I'm guessing that will be handled in a separate PR?

Yuppers! I decided to do that *after* the reorg to make linking easier. It's in my to-do list.

>In the installation bucket, you put Fleet before standalone. I think it also makes sense to have Centrally manage Elastic Agents in Fleet come before Elastic Agent standalone configuration. 

Good point. I'll change the order. My brain wanted to have the installation and config together because the install topic really depends on that info, but I think logically your suggestion makes sense. At some point, we might decide to have a separate top-level bucket for all the standalone content, but not now.

>At the end of the Install standalone Elastic Agent topic, there's a link: Refer to Installation layout for the location of installed Elastic Agent files. This link sends users to the Fleet install topic. I think this could be confusing. What if we used includes to have the installation layout topic in both files?

Good catch. I have a TODO item in the file to remind me to create a similar file for standalone.

>Similarly, at the end of the Install standalone Elastic Agent topic, there's a link to the Elastic Agent standalone configuration docs as a next step. I think a similar link to the Fleet content would be useful at the end of the Fleet install topic.

Good point! I will add it.

File: docs/en/ingest-management/getting-started-traces.asciidoc [hunk 1]
```diff
@@ -61,11 +60,16 @@ In a later step, you'll instrument your code with APM agents and send the data
...
 
-include::elastic-agent/elastic-agent.asciidoc[tag=agent-install-intro]
+{agent} is a single, unified agent that you can deploy to hosts or containers to
+collect data and send it to the {stack}. Behind the scenes, {agent} runs the
+{beats} shippers or Elastic Endpoint required for your configuration.
 
 If you plan on enabling Real User Monitoring (RUM), you must run {agent} centrally.
 If RUM is disabled, you should run {agent} on edge machines.
 
+//QUESTION FOR BRANDON: Should we use an include here like we do in the logs
```
Comment (dedemorton) [inline docs/en/ingest-management/getting-started-traces.asciidoc:70]: @bmorelli25 Note my question here. I think originally we decided to keep things simple by duplicating content, but this section changes too much. It would be better to pull the steps from the official installation docs (like I do in the other QS). I've cleaned up other duplicated info so that we only need to maintain this content in one place now. Let me know what you think.

Comment (bmorelli25) [reply docs/en/ingest-management/getting-started-traces.asciidoc:70]: Ah, I missed this. Yes! Agreed. It made sense to duplicate in the beginning, but no longer. Let's do it.

**Comment by dedemorton** [comment]
The content changes are ready for a review now, too.

**Comment by mostlyjason** [comment]
Looks great! Have you thought about moving the Fleet API under the section "Centrally manage Elastic Agents in Fleet". 

Where did you put https://www.elastic.co/guide/en/fleet/current/secure-connections.html? I couldn't seem to find it but I'm not sure where to look.

Also, have you thought about moving Command Reference under the Elastic Agent section? Its not clear from the title alone that those commands only apply to Elastic Agent.

**Comment by dedemorton** [comment]
@mostlyjason 

> Have you thought about moving the Fleet API under the section "Centrally manage Elastic Agents in Fleet".

Yah that makes a lot of sense. I will move it.

> Where did you put https://www.elastic.co/guide/en/fleet/current/secure-connections.html? I couldn't seem to find it but I'm not sure where to look.

I put it under "Install Fleet-managed Elastic Agents" because it seems like something you need to do as part of installation. https://observability-docs_1109.docs-preview.app.elstc.co/guide/en/fleet/master/secure-connections.html

I decided not to create a separate security container (at least not yet) since we don't have conceptual info that explains the security model and such.

> Also, have you thought about moving Command Reference under the Elastic Agent section? Its not clear from the title alone that those commands only apply to Elastic Agent.

Good point. I will move that one too.

**Review by bmorelli25 — APPROVED** [review]
This looks great!

**Review by nimarezainia — APPROVED** [review]
